WcaOnRails/app/models/competition.rb

```diff
@@ -156,6 +156,59 @@ def user_cannot_demote_themself
     end
   end
 
+  attr_accessor :receive_registration_emails
+  def receive_registration_emails=(r)
+    @receive_registration_emails = ActiveRecord::Type::Boolean.new.type_cast_from_database(r)
+  end
+
+  def receiving_registration_emails?(user_id)
+    competition_delegate = competition_delegates.find_by_delegate_id(user_id)
+    if competition_delegate && competition_delegate.receive_registration_emails
+      return true
+    end
+    competition_organizer = competition_organizers.find_by_organizer_id(user_id)
+    if competition_organizer && competition_organizer.receive_registration_emails
+      return true
+    end
+
+    return false
+  end
+
+  def can_receive_registration_emails?(user_id)
+    competition_delegate = competition_delegates.find_by_delegate_id(user_id)
+    if competition_delegate
+      return true
+    end
+    competition_organizer = competition_organizers.find_by_organizer_id(user_id)
+    if competition_organizer
+      return true
+    end
+
+    return false
+  end
+
+  validate :must_be_an_organizer_or_delegate_to_set_receive_registration_emails
+  def must_be_an_organizer_or_delegate_to_set_receive_registration_emails
```

Comment:
Actually, wait. This is here specifically to get a nice error message if someone tries to send this virtual field when they aren't a delegate or an organizer for the competition. It's not a security thing, as `update_receive_registration_emails` simple can't do anything if it discovers that you're neither a delegate nor an organizer.

In practice, I never expect to see this error message, as in our ERB code, we don't even generate the checkbox for `receive_registration_emails` if the current user cannot `can_receive_registration_emails?`.